motion_tracker/utils/image_cropping.py

```diff
@@ -0,0 +1,254 @@
+
+    box_after_crop = update_box_after_crop(box_coords, cropped_area_coords)
+    return cropped_area_coords, box_after_crop
+
+def read_bbox_data(raw_image_dir, parsed_bb_path='work/imagenet/parsed_bb.csv'):
+    '''Return df with data about images and bounding boxes.
+
+    Data captured corresponds to xml files obtained from ImageNet.
+    '''
+
+    # paper says they use .66, but they use search area that is twice bounding box
+    # So, 0.5 may be more appropriate.
+    max_box_frac_of_width =  0.66
+    max_box_frac_of_height = 0.66
+    images_successfully_downloaded = set(os.listdir('data/imagenet/images'))
+    bbox_df = (pd.read_csv('work/imagenet/parsed_bb.csv')
+                    .assign(box_height = lambda df: df.ymax - df.ymin,
+                            box_width = lambda df: df.xmax - df.xmin)
+                    .assign(box_frac_of_height = lambda df: df.box_height / df.height,
+                            box_frac_of_width = lambda df:  df.box_width / df.width)
+                    .query('filename in @images_successfully_downloaded')
+                    .query('box_frac_of_height < @max_box_frac_of_height')
+                    .query('box_frac_of_width < @max_box_frac_of_width'))
+    return bbox_df
+
+def crop_and_resize(img, img_coords, box_coords, output_width, output_height, random_crop=True):
+    '''Return image and the bounding box after a possibly random crop and resize
+
+    Args:
+    -----
+    img: ndarray of image (consistent with opencv format)
+    img_coords: Coordinates of image boundaries
+    box_coords: Bounding box coordinates on img
+    output_width: Desired output width for resizing
+    output_height: Desired output height for resizing
+    random_crop: If True, create stochastic box to determine crop coords.
+                 If False, create crop based on true bounding box
+                 Cropped area is twice as large (and centered) on box either way
+    '''
+
+    crop_coords, box_after_crop = get_crop_coords(box_coords, img_coords, random_crop)
+    cropped_img = img[crop_coords.y0:crop_coords.y1,
+                      crop_coords.x0:crop_coords.x1]
+    box_after_crop = update_box_after_crop(box_coords, crop_coords)
+    final_img = cv2.resize(cropped_img, (output_width, output_height))
+    final_box_coords = update_box_after_resize(box_after_crop, crop_coords,
+                                               output_width, output_height)
+    return final_img, final_box_coords
+
+def imagenet_generator(batch_size=50):
+    '''Generator yielding dictionary. Each dictionary is array of data for model
+
+    batch_size: Number of items in list to be returned.
+    '''
+    raw_image_dir = 'data/imagenet/images/'
+    img_metadata = read_bbox_data(raw_image_dir)
+    output_width = 256
+    output_height = 256
+
+    start_img_data = []
+    start_box_data = []
+    next_img_data = []
+    next_box_data = []
+
+    while True:
+        batch_df = img_metadata.sample(batch_size)
```

Comment:
Will we still do a `sample` like this when we are training? I imagined effectively iterating over all of the training images after randomly shuffling them, or something like that. I getting rid of this `sample` might speed things up as well. The code under the hood of that is somewhat involved, and I could imagine that iterating over the training images will add some small gains.